src/core/control/jobs/ExportTemplate.cpp

```diff
@@ -0,0 +1,193 @@
+#include "ExportTemplate.h"
+
+#include <cmath>  // for round
+#include <utility>
+
+#include "control/jobs/ProgressListener.h"  // for ProgressListener
+#include "model/Document.h"                 // for Document
+#include "model/XojPage.h"                  // for XojPage
+#include "util/i18n.h"                      // for _
+#include "view/DocumentView.h"              // for DocumentView
+
+ExportTemplate::ExportTemplate(Document* doc, fs::path filePath): doc{doc}, filePath{std::move(filePath)} {
+    ElementRangeVector fullRange;
+    fullRange.emplace_back(0, doc->getPageCount() - 1);
+    setExportRange(fullRange);
+}
+
+ExportTemplate::~ExportTemplate() {}
+
+void ExportTemplate::setExportBackground(const ExportBackgroundType exportBackground) {
+    this->exportBackground = exportBackground;
+}
+
+void ExportTemplate::setProgressListener(ProgressListener* progressListener) {
+    this->progressListener = progressListener;
+}
+
+void ExportTemplate::setExportRange(const PageRangeVector& exportRange) {
```

Comment:
Is this used anywhere else than in the constructor?

Reply:
The function is used in `CustomExportJob` & `ExportHelper`. However, I removed its superfluous counter-part `ExportTemplate::setExportRange(const char* range)` in 64826c35148efd746954025773acb4da73f49d6e.

`CustomExportJob` receives the already parsed Range from `ExportDiaglog` and passes it to `ExportTemplate::setExportRange(const PageRangeVector& exportRange)`.

`ExportHelper` receives the raw, not-parsed range. This lead to the confusing addition of `ExportTemplate::setExportRange(const char* range)`.